src/spikeinterface/postprocessing/template_metrics.py

```diff
@@ -290,10 +319,30 @@ def _compute_metrics(self, sorting_analyzer, unit_ids=None, verbose=False, **job
         return template_metrics
 
     def _run(self, verbose=False):
-        self.data["metrics"] = self._compute_metrics(
-            sorting_analyzer=self.sorting_analyzer, unit_ids=None, verbose=verbose
+
+        delete_existing_metrics = self.params["delete_existing_metrics"]
+        metrics_to_compute = self.params["metrics_to_compute"]
```

Comment:
@chrishalcrow one last comment. IMO the `metrics_to_compute` is not really a `param`, but rather a run-time thing. Would it make sense to move the logic here?  As a user, seeing that `metrics_to_compute` parameter in the json file would make me feel weird!

Reply:
Yes, what I'm saying is since the metrics to compute matter when it's time to run, why not just move the code here to avoid having to save the `metrics_to_compute` parameter? We have access here to metric names and delete existing, so we have everything we need!

Reply:
I had a go but it’s not so simple. We need to set the `metric_names` in `set_params`, since this is used for the metadata and in other places (like in `get_data`). So the `params` dict needs to know all the metric which are gonna be contained in the extension: everything that has been and will be computed.

So we have access to what’s already been computed, in the previous extensions `params`, and the complete list of metrics in self.params["metric_names"]. Using this, we can’t distinguish between:
1. a metric which the user included in `metric_names` and was already computed. We want to recompute this because the user has explicitly specified it.
2. a metric which the user has _not_ specified in `metric_names` and was already computed. We don’t want to recompute this because it was already computed.

So we can either
- Change the behavior so that we don’t re-compute metrics that the user specifies and exists already (I think this is bad: the user might be re-comuting because a metric was updated/fixed, and this wouldn’t overwrite the old result, even when explicitly included by the user)
- Give `metrics_to_compute` a better name in params. Something like a catchier version of “metrics_computed_on_most_recent_run`?
- Keep it as is.
- Be cleverer than I've been so far???

Reply:
I see! Makes total sense. Let's leave it as is then and thanks for explaining.

From my comment I didn't consider the option that the user is passing a metric name that is already computed in order to recompute...